mne/gui/_coreg.py

```diff
@@ -903,6 +905,7 @@ def _update_projection_surface(self):
             tris=self.coreg._bem_low_res["tris"],
             nn=self.coreg._bem_low_res["nn"]
         )
+        self._check_inside = _CheckInside(self._head_geo)
```

Comment:
Good idea! Instantiating this once should make it a *lot* faster. The only caveat is that you probably need to change it when the `_head_geo` changes (dense or not).

At one point I think we maybe *always* used the non-dense head for these computations. It definitely makes things faster, but it does mess up things like the tip of the nose. So maybe we should just change it with the head surface changes...

Reply:
> Instantiating this once should make it a lot faster.

That's what I thought at first but then I realized maybe most of the computation is not done in `_CheckInside.__init__()` but in `_CheckInside.__call__()`? I'll keep digging.